zone/aggro.cpp

```diff
@@ -436,11 +436,16 @@ bool Mob::CheckWillAggro(Mob *mob) {
 		pet_owner->IsClient() &&
 		(
 			!RuleB(Aggro, AggroPlayerPets) ||
-			pet_owner->CastToClient()->GetGM()
+			pet_owner->CastToClient()->GetGM() ||
+			mob->GetSpecialAbility(IMMUNE_AGGRO)
 		)
 	) {
 		return false;
 	}
+	
+	if (this->IsNPC() && mob->IsNPC() && mob->GetSpecialAbility(IMMUNE_AGGRO_NPC)) {
```

Comment:
`this` is implied and can be removed here.

Reply:
Oh derp, removed